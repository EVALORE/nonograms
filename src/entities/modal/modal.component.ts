import { BaseComponent, div, h5 } from '@control.ts/min';

export class ModalComponent extends BaseComponent<HTMLDivElement> {
  private readonly modalWrapper = div({ className: 'modal-background' });
  private readonly modalContent: HTMLDivElement;

  constructor(context: string, title: string, content: string | HTMLElement) {
    super({});
    this.modalContent = div(
      { className: `${context}-modal` },
      h5({ txt: title }, typeof content === 'string' ? div({ txt: content }) : content),
    );

    this.appendChildren([this.modalWrapper, this.modalContent]);
  }
}
