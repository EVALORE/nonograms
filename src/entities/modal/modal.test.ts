import { beforeEach, describe, expect, it } from 'vitest';
import { ModalComponent } from './modal.component';

describe('test modal component', () => {
  let modal: ModalComponent;

  beforeEach(() => {
    modal = new ModalComponent('modal', 'title', 'content');
  });

  it('should render the modal component with string content', () => {
    const expectedContent =
      '<div class="modal-background"></div><div class="modal-modal"><h5>title</h5><div>content</div></div>';
    expect(modal.node.innerHTML).toBe(expectedContent);
  });

  it('should match the snapshot', () => {
    expect(modal.node).toMatchSnapshot();
  });
});
