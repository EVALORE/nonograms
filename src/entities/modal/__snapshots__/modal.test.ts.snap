// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test modal component > should match the snapshot 1`] = `
<div>
  <div
    class="modal-background"
  />
  <div
    class="modal-modal"
  >
    <h5>
      title
    </h5>
    <div>
      content
    </div>
  </div>
</div>
`;
